src/parsing/binast.h

```diff
@@ -1367,12 +1374,60 @@ class BinAstThisExpression final : public BinAstExpression {
   BinAstThisExpression() : BinAstExpression(kNoSourcePosition, kThisExpression) {}
 };
 
+// Common supertype for ObjectLiteralProperty and ClassLiteralProperty
+class BinAstLiteralProperty : public ZoneObject {
+ public:
+  BinAstExpression* key() const { return key_and_is_computed_name_.GetPointer(); }
+  BinAstExpression* value() const { return value_; }
+
+  bool is_computed_name() const {
+    return key_and_is_computed_name_.GetPayload();
+  }
+  bool NeedsSetFunctionName() const {
+    return is_computed_name() && (value_->IsAnonymousFunctionDefinition() ||
+                                  value_->IsConciseMethodDefinition() ||
+                                  value_->IsAccessorFunctionDefinition());
+  }
+
+ protected:
+  BinAstLiteralProperty(BinAstExpression* key, BinAstExpression* value, bool is_computed_name)
+      : key_and_is_computed_name_(key, is_computed_name), value_(value) {}
+
+  PointerWithPayload<BinAstExpression, bool, 1> key_and_is_computed_name_;
+  BinAstExpression* value_;
+};
+
+class BinAstObjectLiteralProperty final : public BinAstLiteralProperty {
+ public:
+  ObjectLiteralProperty::Kind kind() const { return kind_; }
```

Comment:
parser-base uses the ObjectLiteralProperty enum, so it would have required a few more changes to break this out. it doesn't seem like we really need to though? open to suggestions here though

Reply:
I think that's fine for now!